Backend

molbazsa opened the pull request:
I think we should merge the Linear Regression code to main so that everything is up to date.

SourCocktail committed:
wip

SourCocktail committed:
fixed Numerical.Arrange

SourCocktail committed:
Added new functions to Numerical.cs

SourCocktail committed:
fixed Numerical.Arrange .2

molbazsa committed:
Create Neuron

molbazsa committed:
Create Sigmoid activation function

molbazsa commented:
Neuron created

molbazsa commented:
Activation functions created:
- Sigmoid
- Tanh
- ReLU

molbazsa committed:
Create Tanh activation function

molbazsa committed:
Create ReLU activation function

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/DataProvider/Numerical.cs:
```diff
@@ -16,104 +22,223 @@ public static class Numerical
+                SortArray(matrix_array);
+                if (matrix_array.Length % 2 == 0)
+                {
+                    double egy = matrix_array[(int)middle - 1];
```
Use english variable name

SzilvasiPeter replied:
Use english variable name

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/DataProvider/Numerical.cs:
```diff
@@ -16,104 +22,223 @@ public static class Numerical
-
-            return result;
+            double[] matrix_array = new double[matrix.GetLength(0)];
+            double middle = matrix_array.GetLength(0) / 2;
```
Change `middle` type from double to int

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/DataProvider/Numerical.cs:
```diff
@@ -16,104 +22,223 @@ public static class Numerical
+                    return (matrix_array[(int)middle]);
+                }
+            }
+            else
```
Remove code duplication

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/DataProvider/Numerical.cs:
```diff
@@ -16,104 +22,223 @@ public static class Numerical
-            return result;
+            double mean = Numerical.Mean(matrix,axis);
+            double result=0;
+            //double[] matrix_array = new double[matrix.GetLength(0)];
```
Remove zombie code

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/DataProvider/Numerical.cs:
```diff
@@ -16,104 +22,223 @@ public static class Numerical
         {
-            double[] array = null;
-
+            decimal stepD = (decimal)step;
```
Change decimal type to double

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/MLModel/IMLModel.cs:
```diff
@@ -7,14 +7,14 @@ namespace LinearRegressionBackend.MLModel
         /// <param name="epochs">The number of iterations.</param>
         /// <returns>Returns with the training history.</returns>
-        History Fit(double[][] inputData, double[] targetData, int epochs=1);
+       // History Fit(double[][] inputData, double[] targetData, int epochs=1);
```
Remove zombie code. Place summary to `Train` method.

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/MLNeuralNetwork/Neuron.cs:
```diff
@@ -0,0 +1,26 @@
+        private double _Bias;
+        private IActivationFunction _ActivationFunction;
+
+        public Neuron(Vector<double> weights, IActivationFunction activationFunction) {
```
Initialize bias as well

SzilvasiPeter commented on LinearRegression/LinearRegressionBackend/Program.cs:
```diff
@@ -35,7 +35,47 @@ static void Main(string[] args)
+
+            double testDongle =Numerical.StandardDeviation(myArray4,1);
+            Console.WriteLine(testDongle);
+            /*
```
Remove zombie code

SzilvasiPeter commented on LinearRegression/Numrical_uTest/NumericalTest.cs:
```diff
@@ -0,0 +1,203 @@
+        {
+            // Arrange
+            double[][] testMatrix = new double[5][];
+            testMatrix[0] = new double[2];
```
Simplify `testMatrix` initialization.

SzilvasiPeter reviewed (commented):
Review findings were added

molbazsa committed:
Make Neuron properties public

The weights, bias and activation function properties were made public
so that the UI can interact with them

molbazsa committed:
Create Neuron and IActivationFunction tests

molbazsa committed:
Tidy Numerical

molbazsa committed:
Tidy IMLModel
